paper/paper.tex

```diff
@@ -344,70 +422,81 @@ \subsubsection{OpenFOAM}
 \subsubsection{SU2}
 
 The SU2-preCICE adapter~\cite{Uekermann2017_Adapters} patches and extends the original CFD code SU2~\cite{su2}.
-To this end, the adapter provides a script to run before the build phase.
-We do this in Nix' patch phase. Afterwards, \texttt{stdenv} automatically recognizes that SU2 uses autotools for building and runs the configure phase.
+To this end, the adapter provides a script to run before the \texttt{buildPhase}.
+We do this in Nix' \texttt{patchPhase}, which is responsible for patching the source before running the \texttt{buildPhase}. Afterwards, \texttt{stdenv} automatically recognizes that SU2 uses Autotools for building and runs the \texttt{configurePhase}.
 Finally, to find the preCICE installation, we need to set the configuration flags \texttt{--with-include} and \texttt{--with-lib}.
 
+Since preCICE distribution v202404, this adapter is replaced by a python-based solution
+
 \subsection{preCICE VM}
 
 The current preCICE VM\footnote{\url{https://precice.org/installation-vm.html}} is built on top of Vagrant\footnote{\url{https://www.vagrantup.com/}} using Ubuntu as a base image.
-When run for the first time, it further installs software, compiles programs, and clones the preCICE tutorials\footnote{\url{https://github.com/precice/tutorials/}}.
-This inherently breaks reproducibility as the VM fetches files from the internet that are at this very point in time the \textit{latest} version, e.g. the \texttt{main} branch of the preCICE repository.
+When provisioned for the first time, it further installs software, compiles programs, and clones the preCICE tutorials\footnote{\url{https://github.com/precice/tutorials/}}.
+This means that we only have reproducibility when fetching the VM from vagrant cloud~\footnote{\url{https://app.vagrantup.com/precice/boxes/precice-vm}}, as we can not rebuild the VM from scratch, because the provision step currently fetches files from the internet that are at this very point in time the \textit{latest} version, i.e. the main branch of the preCICE repository rather than a pinned tag or commit.
+Making us dependent on vagrant cloud to continue to serve the box.
```

Comment:
This is better than before but, just for the sake of discussion, I think the main issue here that prevents reproducibility is not fetching the latest versions (this is just a matter of which versions are specified in the provisioning scripts).

Instead, an actual issue is that the Ubuntu repositories (and maybe other related repositories) eventually become unavailable.

Also, the box does not need to be hosted on Vagrant Cloud, but it can also be shared on via any shared file system or similar. It is just one (very big) file.

An issue here in the future might be that the specific vagrant box file (which is prepared for a specific major version of VirtualBox) is not compatible with future versions of VirtualBox or Vagrant itself.